R/crew_launcher_lsf.R

```diff
@@ -0,0 +1,287 @@
+      super$initialize(
+        name = name,
+        seconds_launch = seconds_launch,
+        seconds_interval = seconds_interval,
+        seconds_timeout = seconds_timeout,
+        seconds_idle = seconds_idle,
+        seconds_wall = seconds_wall,
+        seconds_exit = seconds_exit,
+        tasks_max = tasks_max,
+        tasks_timers = tasks_timers,
+        reset_globals = reset_globals,
+        reset_packages = reset_packages,
+        reset_options = reset_options,
+        garbage_collection = garbage_collection,
+        verbose = verbose,
+        command_submit = command_submit,
+        command_delete = command_delete,
+        script_directory = script_directory,
+        script_lines = script_lines
+      )
+      self$lsf_log_output <- lsf_log_output
+      self$lsf_log_error <- lsf_log_error
+      self$lsf_memory_megabytes_per_cpu <- lsf_memory_megabytes_per_cpu
+      self$lsf_cpus_per_task <- lsf_cpus_per_task
+    },
+    #' @description Validate the launcher.
+    #' @return `NULL` (invisibly). Throws an error if a field is invalid.
+    validate = function() {
+      super$validate()
+      fields <- c("lsf_log_output", "lsf_log_error")
+      for (field in fields) {
+        if (!is.null(self[[field]])) {
+          crew::crew_assert(
+            self[[field]],
+            is.character(.),
+            length(.) == 1L,
+            !anyNA(.),
+            nzchar(.),
+            message = paste(
+              field,
+              "must be either NULL or a nonempty length-1 character string."
+            )
+          )
+        }
+      }
+      fields <- c(
+        "lsf_memory_megabytes_per_cpu",
+        "lsf_cpus_per_task"
+      )
+      for (field in fields) {
+        if (!is.null(self[[field]])) {
+          crew::crew_assert(
+            self[[field]],
+            is.numeric(.),
+            length(.) == 1L,
+            !anyNA(.),
+            . > 0L,
+            message = paste("invalid", field, "field")
+          )
+        }
+      }
+      invisible()
+    },
+    #' @description Generate the job script.
+    #' @details Includes everything except the worker-instance-specific
+    #'   job name and the worker-instance-specific
+    #'   call to `crew::crew_worker()`, both of which get inserted at
+    #'   the bottom of the script at launch time.
+    #' @return Character vector of the lines of the job script.
+    #' @param name Character of length 1, name of the job. For inspection
+    #'   purposes, you can supply a mock job name.
+    #' @examples
+    #' launcher <- crew_launcher_lsf(
+    #'   lsf_log_output = "log_file_%J.log",
+    #'   lsf_log_error = NULL,
+    #'   lsf_memory_megabytes_per_cpu = 4096
+    #' )
+    #' launcher$script(name = "my_job_name")
+    script = function(name) {
+      c(
+        "#!/bin/sh",
+        paste("#BSUB-J", name),
+        if_any(
+          is.null(self$lsf_log_output),
+          character(0L),
+          paste("#BSUB-o", self$lsf_log_output)
```

Comment:
Again, I wonder if we need a space between the "#BSUB" comment and flags like "-o". Might be nice to have for readability if job submission will still work.